### PR TITLE
Fix method tests to work with inplace special ops

### PR DESCRIPTION
**Pull request by xoiga123**
[Trello task](https://trello.com/c/FjGCje7g/1291-fix-method-tests-to-work-with-inplace-special-ops)

### CODE REVIEW

**Comment by xoiga123** [comment]
@VedPatwardhan do you mind taking a look at `torch.matmul` inplace update as discussed in discord? :dizzy_face:

File: ivy/functional/backends/torch/linear_algebra.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def matmul(
     adjoint_b: bool = False,
     out: Optional[torch.Tensor] = None,
 ) -> torch.Tensor:
-
+    x1, x2 = [x.clone() if out is x else x for x in (x1, x2)]
```
Comment (xoiga123) [inline ivy/functional/backends/torch/linear_algebra.py:171]: @VedPatwardhan

**Comment by xoiga123** [comment]
@fspyridakos :sweat_smile:

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 2]
```diff
@@ -1115,8 +1116,8 @@ def test_method(
         for v, d in zip(init_flags.as_variable, init_input_dtypes)
     ]
 
-    # Create Args
-    args_constructor, kwargs_constructor, *_ = create_args_kwargs(
+    # Save original constructor data for inplace operations
+    con_data = OrderedDict(
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:1120]: Is it necessary to create an OrderedDict here? can't we just copy the original args_constructor and kwargs_constructor?

File: ivy_tests/test_ivy/test_misc/test_array.py [hunk 3]
```diff
@@ -995,6 +1003,10 @@ def test_array__truediv__(
     on_device,
 ):
     dtype, x = dtype_and_x
+
+    # bfloat16 is not supported by numpy
+    assume(not ("bfloat16" in dtype))
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_misc/test_array.py:1008]: We shouldn't be skipping specific dtypes, this ideally shouldn't be generated for NumPy backend anyways.

File: ivy_tests/test_ivy/test_misc/test_array.py [hunk 4]
```diff
@@ -977,12 +986,11 @@ def test_array__rdivmod__(
 @handle_method(
     method_tree="Array.__truediv__",
     dtype_and_x=helpers.dtype_and_values(
-        available_dtypes=helpers.get_dtypes("numeric"),
+        available_dtypes=helpers.get_dtypes("real_and_complex"),
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_misc/test_array.py:989]: Why are we limiting the data type generation to real_and_complex?

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers/function_testing.py (hunk 2 above)
Comment (xoiga123) [reply ivy_tests/test_ivy/helpers/function_testing.py:1120]: https://github.com/unifyai/ivy/blob/74d08c39bb607eb09b020146388e7c0bfceceb3b/ivy_tests/test_ivy/helpers/function_testing.py#L1219-L1222

args_constructor and kwargs_constructor may not be ordered (I don't think our Container is ordered in the early python3s? It only inherits from Dict), and later we may have to pass the arguments in the correct order to another process.

File: ivy_tests/test_ivy/test_misc/test_array.py (hunk 4 above)
Comment (xoiga123) [reply ivy_tests/test_ivy/test_misc/test_array.py:989]: real_and_complex is numeric + complex so it's widening the data type

File: ivy_tests/test_ivy/test_misc/test_array.py (hunk 3 above)
Comment (xoiga123) [reply ivy_tests/test_ivy/test_misc/test_array.py:1008]: My bad, I'll look into this, `asarray` is called with torch backend somewhere and torch.as_tensor really said `TypeError: can't convert np.ndarray of type bfloat16.` :skull:

**Comment by fspyridakos** [comment]
Sorry @xoiga123, was looking into it but was taking my time to fully go over it. Could you sync your brach with upstream to undo the git force push commits?

**Pull request closed without merging**